src/types.ts

```diff
@@ -67,12 +67,14 @@ export interface DirectusProviderProps {
 /**
  * Shape of the main context.
  */
-export interface DirectusContextType {
+export interface DirectusContextType<T extends TypeMap> {
   apiUrl: string;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  directus: IDirectus<any>;
+  directus: IDirectus<T>;
```

Comment:
Can T = any by default, so legacy code doesn't to be changed? If we leave IDirectus<T> people is forced to update the code, right?

Reply:
I can set a `TypeMap` as the type for the Directus provider, so legacy code won't be afected in any way.